src/main/java/fi/aalto/cs/apluscourses/ui/base/TreeView.java

```diff
@@ -56,6 +59,17 @@ public TreeView() {
     addMouseListener(new TreeMouseListener());
   }
 
+  /**
+   * Sets the {@link JPopupMenu} that is shown for elements.
```

Comment:
This isn't the most descriptive comment is it

Reply:
Reminds me of
```
/**
 * Constructor.
 */
```
But in this case I think the comment is completely fine, since the method name is pretty obvious :+1: Maybe the only addition would be that the pop-up menu is shown on right click? Although maybe that is obvious as well :smile:

Reply:
Found it: https://github.com/Aalto-LeTech/intellij-plugin/blob/c0a052474adbd444fde010fca01677612c1c8d60/src/main/java/fi/aalto/cs/apluscourses/model/ComponentInstallerImpl.java#L21